ops.qmd

```diff
@@ -2,84 +2,759 @@
 
+This chapter explores the practices and architectures needed to effectively develop, deploy, and manage ML models across their entire lifecycle. We examine the various phases of the ML process including data collection, model training, evaluation, deployment, and monitoring. The importance of automation, collaboration, and continuous improvement is also discussed. We contrast different environments for ML model deployment, from cloud servers to embedded edge devices, and analyze their distinct constraints. Through concrete examples, we demonstrate how to tailor ML system design and operations for reliable and optimized model performance in any target environment. The goal is to provide readers with a comprehensive understanding of ML model management so they can successfully build and run ML applications that sustainably deliver value.
+
 ::: {.callout-tip}
 ## Learning Objectives
 
-* coming soon.
+* Understand what is MLOps and why it is needed
+
+* Learn the architectural patterns for traditional MLOps
+
+* Contrast traditional vs. embedded MLOps across the ML lifecycle 
+
+* Identify key constraints of embedded environments 
+
+* Learn strategies to mitigate embedded ML challenges
+
+* Examine real-world case studies demonstrating embedded MLOps principles
+
+* Appreciate the need for holistic technical and human approaches 
 
 :::
 
 ## Introduction
 
-Explanation: This subsection sets the groundwork for the discussions to follow, elucidating the fundamental concept of MLOps and its critical role in enhancing the efficiency, reliability, and scalability of embedded AI systems. It outlines the unique characteristics of implementing MLOps in an embedded context, emphasizing its significance in the streamlined deployment and management of machine learning models.
+Machine Learning Operations (MLOps), is a systematic approach that combines machine learning (ML), data science, and software engineering to automate the end-to-end ML lifecycle. This includes everything from data preparation and model training to deployment and maintenance. MLOps ensures that ML models are developed, deployed, and maintained efficiently and effectively.
+
+Consider a ridesharing company that wants to deploy a machine-learning model to predict rider demand in real time. The data science team spends months developing a model, but when it's time to deploy, they realize it needs to be compatible with the engineering team's production environment. Deploying the model requires rebuilding it from scratch - costing weeks of additional work. This is where MLOps comes in. 
```

Comment:
This is a small thing but opening with a non-edge ML example feels slightly disconnected from the rest of the book. I don't think you would need to change it but I might just add a sentence specifying that it's not an edge model. specifically mention where "production environment" is

Reply:
Good point, fixed it. Thanks @alxrod